File: SDKs/probely.com/1.2.0/python/src/sdk/models/operations/post_setpassword_.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class PostSetpasswordRequestBody:
    confpassword: dict[str, Any] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('confpassword') }})
    password: dict[str, Any] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('password') }})
    token: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('token') }})
    

@dataclass_json
@dataclasses.dataclass
class PostSetpassword200ApplicationJSON:
    message: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('message') }})
    

@dataclass_json
@dataclasses.dataclass
class PostSetpassword400ApplicationJSON:
    error: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('error') }})
    

@dataclasses.dataclass
class PostSetpasswordRequest:
    request: PostSetpasswordRequestBody = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class PostSetpasswordResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    post_setpassword_200_application_json_object: Optional[PostSetpassword200ApplicationJSON] = dataclasses.field(default=None)
    post_setpassword_400_application_json_object: Optional[PostSetpassword400ApplicationJSON] = dataclasses.field(default=None)
    

File: SDKs/probely.com/1.2.0/typescript/src/sdk/models/operations/postsetpassword.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class PostSetpasswordRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=confpassword" })
  confpassword: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=password" })
  password: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=token" })
  token: string;
}


export class PostSetpassword200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=message" })
  message?: string;
}


export class PostSetpassword400ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=error" })
  error?: string;
}


export class PostSetpasswordRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: PostSetpasswordRequestBody;
}


export class PostSetpasswordResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  postSetpassword200ApplicationJSONObject?: PostSetpassword200ApplicationJson;

  @SpeakeasyMetadata()
  postSetpassword400ApplicationJSONObject?: PostSetpassword400ApplicationJson;
}
